### DISCUSSION TITLE
Is it possible to open a directory based on an HTTP header?

### DISCUSSION THREAD

**Discussion by russellseymour** [eclipse-theia/theia#8116] [General]
I have put an `oauth2_proxy` in front of my Theia container, and it is configured to send user headers, such as `X-Forwarded_User` to the container.

Theia has a a volume mapped to it containing directories of usernames in which people will work on their projects, e.g:

```
 - john
 - paul
 - ringo
 - george
```

The proxy acts as reverse proxy and when a user is authenticated it proxies requests to the theia container.

What I was surprised to see was that if I access the proxy on `http://localhost:4180` and have the upstream set as `http://localhost:8080/#/home/project` that Theia would append the `/#/home/project` to the URL in the browser - I had assumed the the URL would not be changed as I have this set in the upstream configuration.

Anyway, to the actual question. Is there a way to extend the directory that is opened based on the HTTP header that has been sent. For example if `john` logs in then I want the URL to state `/#/home/project/john`. I know this is not any form of isolation as someone could just retype the URL, but I want people to drop into their own directory to begin with.

I have tried to put in an NGinx proxy as well and set the `ProxyPassReverse` configuration to `http://localhost:3000/#/home/project/$http_x_forwarded_user` but this gave me a blank page. If I just use the `http://localhost:3000` or `http://localhost:3000/#/home/project` things work and I see the application in the browser.

I know that Theia is not concerned about performing any authentication, but being able to use headers would be a great help or if anyone has any idea how I can configure `ProxyPassReverse` to work with the header that would be acceptable to. :)

**Comment by akosyakov** [comment]
I wonder is there nginx community to ask such questions. It does not seem to related to Theia in particular.

**Reply by russellseymour** [reply]
Yes I plan on asking on an Nginx community forum as well, but this question is related to Theia in that I wanted to know if there was a way to open a directory based on a header coming from Nginx.

As Theia is running as a web service it has access to the headers.

**Reply by akosyakov** [reply]
You can open any folder using `WorkspaceService.open` API via the product specific extension given that you have a header in the frontend. But It sounds that rewriting a result via NGinx should work as well. It probably does not work because of some proxy or network misconfigurations.
